hal/src/gcc/pinmap_impl.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "pinmap_hal.h"
```

Comment:
Is this necessary? `pinmap_hal.h` is supposed to include implementation instead.

Same:
https://github.com/particle-iot/device-os/pull/1838/files#diff-164b14eb9d911cd3e319975e9c752d06R20
https://github.com/particle-iot/device-os/pull/1838/files#diff-0170c416cc4060c26c20782c5a669a4bR3
https://github.com/particle-iot/device-os/pull/1838/files#diff-6505829b97e07fa5b407eb51407c4d08R24

Reply:
Removed.